### DISCUSSION TITLE
Memory leak in @urql/core@2.4.3?

### DISCUSSION THREAD

**Discussion by ghost** [urql-graphql/urql#2904] [Q&A] [answered]
Hi everyone,
Are you familiar with a memory leak issue in `@urql/core` 2.4.3 version?

We have some insights that points to a memory leak caused when using createClient in Node.js. We see retained objects in wonka.js which is a dependency for @urql/core.

Our usage:
```js
// module A
const fetch = require('cross-fetch');
const { createClient } = require('@urql/core');

const financialAnalyticsGraphqlClient = createClient({ url, fetch, requestPolicy: 'network-only' });

module.exports = { financialAnalyticsGraphqlClient };

// module B
await financialAnalyticsGraphqlClient.query(EARNING_THIS_MONTH_QUERY, variables).toPromise()
```

This code is executed when rendering the page to fetch the initial data.

Thanks

**Comment by kitten** [comment, marked as answer]
Can you upgrade past `3.1.0` please and make sure that this bumps your version of `wonka`? This should resolve the issue; see: https://github.com/urql-graphql/urql/blob/main/packages/core/CHANGELOG.md#patch-changes-1

**Reply by ghost** [reply]
Thanks for the quick response! 🎖️ 🙏
